server/server.go

```diff
@@ -137,12 +139,40 @@ func (s *Server) SyncStatus(instID flux.InstanceID, ref string) (res []string, e
 }
 
 // LogEvent stores an event in the instance's history.
-func (s *Server) LogEvent(instID flux.InstanceID, event history.Event) error {
+func (s *Server) LogEvent(instID flux.InstanceID, e history.Event) error {
 	helper, err := s.instancer.Get(instID)
 	if err != nil {
 		return errors.Wrapf(err, "getting instance")
 	}
-	return helper.LogEvent(event)
+
+	// Log event in history first. This is less likely to fail
+	err = helper.LogEvent(e)
+	if err != nil {
+		return errors.Wrapf(err, "logging event")
+	}
+
+	// If this is a release
+	if e.Type == history.EventRelease {
+		// Casting to ReleaseEventMetadata
+		bytes, err := json.Marshal(e.Metadata)
```

Comment:
Prior to this code, the only use of the `Event.Metadata` was when it was converted to `map[string]interface{}` by the HTTP handler, then stored in postgres (after re-marshalling again).

I'm the first person to unmarshall the data because I need to to get the info for the notification.

So there are no other things to unmarshall.